### PR TITLE
Add Drqv2 Parallel Baseline and Benchmark Shell Scripts

### PR DESCRIPTION
**Pull request by YilangLiu**

### CODE REVIEW

File: examples/cfg/drqv2/task/ant.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      activation: elu
+  
+  general:
+    device: cuda:0
+    render: False 
+    seed: 0
+    train: True
+    
+  config:
+    name: df_ant_dva
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/ant.yaml]: Name should not be df_ant_dva but df_ant_drqv2

File: examples/cfg/drqv2/task/cheetah.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      activation: elu
+  
+  general:
+    device: cuda:0
+    render: False 
+    seed: 0
+    train: True
+
+  config:
+    name: df_cheetah_dva
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/cheetah.yaml]: see previous

File: examples/cfg/drqv2/task/ant.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    img_aug: False
+    img_aug_padding: 4
+    save_interval: 400
+
+    player:
+      determenistic: True
+      games_num: 1
+      num_actors: 1
+      print_stats: True
+      render_mode: dmc # ['usd', 'dmc']
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/ant.yaml]: This render_mode is no longer use

File: examples/cfg/drqv2/task/cheetah.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    img_aug: False
+    img_aug_padding: 4
+    save_interval: 400
+
+    player:
+      determenistic: True
+      games_num: 1
+      num_actors: 1
+      print_stats: True
+      render_mode: dmc # ['usd', 'dmc']
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/cheetah.yaml]: see previous

File: examples/cfg/drqv2/task/hopper.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      activation: elu
+  
+  general:
+    device: cuda:0
+    render: False 
+    seed: 0
+    train: True
+
+  config:
+    name: df_hopper_dva
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/hopper.yaml]: see previous

File: examples/cfg/drqv2/task/hopper.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    img_aug: False
+    img_aug_padding: 4
+    save_interval: 400
+
+    player:
+      determenistic: False
+      games_num: 1
+      num_actors: 1
+      print_stats: True
+      render_mode: dmc # ['usd', 'dmc']
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/hopper.yaml]: see previous

File: examples/cfg/drqv2/task/humanoid.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      activation: elu
+  
+  general:
+    device: cuda:0
+    render: False 
+    seed: 0
+    train: True
+  
+  config:
+    name: df_humanoid_dva
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/humanoid.yaml]: see previous

File: examples/cfg/drqv2/task/humanoid.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    img_aug: False
+    img_aug_padding: 4
+    save_interval: 400
+
+    player:
+      determenistic: True
+      games_num: 1
+      num_actors: 1
+      print_stats: True
+      render_mode: dmc # ['usd', 'dmc']
```
Comment (HaoxiangYou) [inline examples/cfg/drqv2/task/humanoid.yaml]: see previous

File: examples/drqv2.py
Comment (HaoxiangYou) [inline examples/drqv2.py:1]: may need to move this to like external/drqv2_parallel or is this file being used at all?

File: dflex/dflex.egg-info/PKG-INFO
Comment (HaoxiangYou) [inline dflex/dflex.egg-info/PKG-INFO:1]: do not change this

File: dflex/dflex.egg-info/SOURCES.txt
Comment (HaoxiangYou) [inline dflex/dflex.egg-info/SOURCES.txt:1]: do not change this

File: examples/scripts.sh [hunk 9]
```diff
@@ -1,6 +1,24 @@
...
+    for algorithm in $algorithms
+    do 
+        for num_actor in $num_actors
+        do 
+            for seed in $seeds
+            do
+                for learning_rate in $learning_rates
+                do
+                    echo "Now Running Experiment: $algorithm, Robot: $env Seed: $seed, Number of Actors: $num_actor, Actor Learning Rate: $learning_rate"
+                    python train_dva.py --cfg ./cfg/$algorithm/$env.yaml --actor_learning_rate $learning_rate --logdir ./logs/$env/$algorithm --seed $seed --num_actors $num_actor --no-time-stamp
```
Comment (HaoxiangYou) [inline examples/scripts.sh]: should this be full_state for dva and for shac ? if dva should the logdir also name full state?

File: examples/scripts.sh
Comment (HaoxiangYou) [inline examples/scripts.sh:1]: should the name of script be changed? like eval_full_statel, you choose a better name?

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/tb/events.out.tfevents.1742845704.yilang-Raider-GE66-12UGS.1296392.0
Comment (HaoxiangYou) [inline externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/tb/events.out.tfevents.1742845704.yilang-Raider-GE66-12UGS.1296392.0:1]: do not include training logs

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/tb/events.out.tfevents.1742840869.yilang-Raider-GE66-12UGS.752312.0
Comment (HaoxiangYou) [inline externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/tb/events.out.tfevents.1742840869.yilang-Raider-GE66-12UGS.752312.0:1]: do not add tensorboard file

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/args.json
Comment (HaoxiangYou) [inline externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/args.json:1]: do not add training log

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/args.json
Comment (HaoxiangYou) [inline externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/args.json:1]: do not add training logs

File: examples/train_script.py [hunk 10]
```diff
@@ -38,6 +38,8 @@
         script_name = 'train_bptt.py'
     elif args.algo == "dva":
         script_name = "train_dva.py"
+    elif args.algo == "dva":
+        script_name == "train_drqv2.py"
```
Comment (HaoxiangYou) [inline examples/train_script.py]: train_drqv2_parallel or train_drqv2?

File: examples/train_script.py [hunk 11]
```diff
@@ -38,6 +38,8 @@
         script_name = 'train_bptt.py'
     elif args.algo == "dva":
         script_name = "train_dva.py"
+    elif args.algo == "dva":
```
Comment (HaoxiangYou) [inline examples/train_script.py]: algo == "drqv2"

File: examples/train_rl.py
Comment (HaoxiangYou) [inline examples/train_rl.py:1]: accidentally change this file? convert it back

File: examples/train_shac.py
Comment (HaoxiangYou) [inline examples/train_shac.py:1]: do you also need to change learning rate here or no? since the scripts.sh include it or you should separate the scripts.sh to dva and shac

File: examples/test_env.py
Comment (HaoxiangYou) [inline examples/test_env.py:1]: could you add this to test_env_drqv2 and change the original back?

File: examples/train_drqv2_parallel.py [hunk 12]
```diff
@@ -0,0 +1,409 @@
...
+                self.writer.add_scalar('rewards/iter', -mean_policy_loss, actor_step)
+            else:
+                mean_policy_loss = np.inf
+                mean_episode_length = 0
+
+            self.writer.flush()
+            time_end_epoch = time.time()
+            print('iter {}: ep loss {:.2f}, ep len {}, fps total {:.2f}'.format(\
+                        episode_step, mean_policy_loss, mean_episode_length, 
+                        self.cfg.action_repeat * self.num_envs / (time_end_epoch - time_start_epoch)))
```
Comment (HaoxiangYou) [inline examples/train_drqv2_parallel.py]: I think probably is better to fps in env_step or no? probably you can add another print for entire procsee, like fps: fps env step:?

File: examples/train_dva.py [hunk 13]
```diff
@@ -67,7 +67,9 @@ def get_args(): # TODO: delve into the arguments
         {"name": "--no-time-stamp", "action": "store_true", "default": False,
             "help": "whether not add time stamp at the log path"},
         {"name": "--device", "type": str, "default": "cuda:0"},
-        {"name": "--seed", "type": int, "default": 0, "help": "Random seed"}]
+        {"name": "--seed", "type": int, "default": 0, "help": "Random seed"},
+        {"name": "--actor_learning_rate", "type": float, "default": 2e-3, "help": "Actor Learning Rate"},
```
Comment (HaoxiangYou) [inline examples/train_dva.py]: would it be possible to change actor learning rate to None as default, and if none use the original config

File: examples/train_dva.py [hunk 14]
```diff
@@ -92,15 +94,25 @@ def get_args(): # TODO: delve into the arguments
 
     if not args.no_time_stamp:
         args.logdir = os.path.join(args.logdir, get_time_stamp())
+
+    if args.num_actors:
+        cfg_train["params"]["config"]["num_actors"] = args.num_actors
+        args.logdir = args.logdir + "/num_actors_"+str(args.num_actors)
+
```
Comment (HaoxiangYou) [inline examples/train_dva.py:102]: if no num_actor, do not change the name?

File: examples/train_dva.py [hunk 15]
```diff
@@ -92,15 +94,25 @@ def get_args(): # TODO: delve into the arguments
...
+    if args.actor_learning_rate:
+        cfg_train["params"]["config"]["actor_learning_rate"] = args.actor_learning_rate
+        args.logdir = args.logdir + "_actor_lr_" + str(args.actor_learning_rate)
```
Comment (HaoxiangYou) [inline examples/train_dva.py]: same here

File: .vscode/settings.json
Comment (HaoxiangYou) [inline .vscode/settings.json:1]: no vscode setting

File: examples/cfg/drqv2/task/ant.yaml (hunk 1 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/ant.yaml]: fixed

File: examples/cfg/drqv2/task/cheetah.yaml (hunk 2 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/cheetah.yaml]: fixed

File: examples/cfg/drqv2/task/ant.yaml (hunk 3 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/ant.yaml]: fixed

File: examples/cfg/drqv2/task/cheetah.yaml (hunk 4 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/cheetah.yaml]: fixed

File: examples/cfg/drqv2/task/hopper.yaml (hunk 5 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/hopper.yaml]: fixed

File: examples/cfg/drqv2/task/hopper.yaml (hunk 6 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/hopper.yaml]: fixed

File: examples/cfg/drqv2/task/humanoid.yaml (hunk 7 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/humanoid.yaml]: fixed

File: examples/cfg/drqv2/task/humanoid.yaml (hunk 8 above)
Comment (YilangLiu) [reply examples/cfg/drqv2/task/humanoid.yaml]: fixed

File: examples/drqv2.py
Comment (YilangLiu) [reply examples/drqv2.py:1]: temporary leave here for Hydra Importing

File: dflex/dflex.egg-info/PKG-INFO
Comment (YilangLiu) [reply dflex/dflex.egg-info/PKG-INFO:1]: fixed

File: dflex/dflex.egg-info/SOURCES.txt
Comment (YilangLiu) [reply dflex/dflex.egg-info/SOURCES.txt:1]: fixed

File: examples/scripts.sh
Comment (YilangLiu) [reply examples/scripts.sh:1]: see previous

File: examples/scripts.sh (hunk 9 above)
Comment (YilangLiu) [reply examples/scripts.sh]: (1) This script will be dva full state only. I have renamed the scripts.sh to scripts_dva_full_state.sh for clarity. (2) The training log directory will be saved in logs/dva/envs_full_state. (3) For other baselines, we will have separate scripts

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/tb/events.out.tfevents.1742845704.yilang-Raider-GE66-12UGS.1296392.0
Comment (YilangLiu) [reply externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/tb/events.out.tfevents.1742845704.yilang-Raider-GE66-12UGS.1296392.0:1]: deleted and add it to .gitignore

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/tb/events.out.tfevents.1742840869.yilang-Raider-GE66-12UGS.752312.0
Comment (YilangLiu) [reply externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/tb/events.out.tfevents.1742840869.yilang-Raider-GE66-12UGS.752312.0:1]: deleted and add it to .gitignore

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/args.json
Comment (YilangLiu) [reply externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s657195-pixel/args.json:1]: deleted and add it to .gitignore

File: externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/args.json
Comment (YilangLiu) [reply externals/curl/tmp/cartpole/cartpole-swingup-03-24-im84-b128-s419636-pixel/args.json:1]: deleted and add it to .gitignore

File: examples/train_script.py (hunk 10 above)
Comment (YilangLiu) [reply examples/train_script.py]: We do not need this python file for running large experiments. Delete this python file to avoid confusion

File: examples/train_script.py (hunk 11 above)
Comment (YilangLiu) [reply examples/train_script.py]: see previous

File: examples/train_rl.py
Comment (YilangLiu) [reply examples/train_rl.py:1]: Fixed

File: examples/train_shac.py
Comment (YilangLiu) [reply examples/train_shac.py:1]: fixed

File: examples/test_env.py
Comment (YilangLiu) [reply examples/test_env.py:1]: fixed, remove test_env for drqv2

File: examples/train_drqv2_parallel.py (hunk 12 above)
Comment (YilangLiu) [reply examples/train_drqv2_parallel.py]: fixed

File: examples/train_dva.py (hunk 13 above)
Comment (YilangLiu) [reply examples/train_dva.py]: fixed. see comments in the train_dva.py

File: examples/train_dva.py (hunk 14 above)
Comment (YilangLiu) [reply examples/train_dva.py:102]: see previous

File: examples/train_dva.py (hunk 15 above)
Comment (YilangLiu) [reply examples/train_dva.py]: see previous
